komga/src/main/kotlin/org/gotson/komga/domain/service/SeriesCollectionLifecycle.kt

```diff
@@ -36,8 +36,15 @@ class SeriesCollectionLifecycle(
 
     if (existing.name != toUpdate.name && collectionRepository.existsByName(toUpdate.name))
       throw DuplicateNameException("Collection name already exists")
+    val allSeriesIds = toUpdate.seriesIds.toMutableList()
+    collectionRepository.findDeletedSeriesByName(toUpdate.name).let { deleted ->
+      deleted.forEach { (number, seriesId) ->
+        if (allSeriesIds.size < number) allSeriesIds.add(seriesId)
+        else allSeriesIds.add(number, seriesId)
+      }
+    }
 
-    collectionRepository.update(toUpdate)
+    collectionRepository.update(toUpdate.copy(seriesIds = allSeriesIds))
```

Comment:
Same as for Read Lists, i don't understand what that bit of code does. Can you explain?